src/systems/material.js

```diff
@@ -63,6 +64,21 @@ module.exports.System = registerSystem('material', {
     textureCache[hash].then(handleImageTextureLoaded);
   },
 
+  /**
+   * High-level function for loading canvas textures (THREE.Texture).
+   *
+   * @param {Element|string} src - Texture source.
+   * @param {object} data - Texture data.
+   * @param {function} cb - Callback to pass texture to.
+   */
+  loadCanvas: function (src, data, cb) {
+    // duck-punch a readyState and a HAVE_CURRENT_DATA attiribute onto the canvas element so THREE.videoTexture will work

```

Comment:
`Hack readyState and HAVE_CURRENT_DATA on canvas to work with THREE.VideoTexture`